packages/core/src/utils/__tests__/MessageValidator.test.ts

```diff
@@ -0,0 +1,49 @@
+import { ConnectionInvitationMessage } from '../../modules/connections'
+import { MessageValidator } from '../MessageValidator'
+
+describe('MessageValidator', () => {
+  describe('validate', () => {
+    it('validates a class instance of ConnectionMessage', () => {
+      const invitation = new ConnectionInvitationMessage({
+        did: 'did:sov:test1234',
+        id: 'afe2867e-58c3-4a8d-85b2-23370dd9c9f0',
+        label: 'test-label',
+      })
+
+      expect(MessageValidator.validate(invitation)).resolves
+    })
+    it('throws an error for an invalid class instance of ConnectionMessage', () => {
+      const invitation = new ConnectionInvitationMessage({
+        did: 'did:sov:test1234',
+        id: 'afe2867e-58c3-4a8d-85b2-23370dd9c9f0',
+        label: 'test-label',
+      })
+
+      invitation.did = 'hello, world!'
+
+      expect(MessageValidator.validate(invitation)).rejects
+    })
+  })
+
+  describe('validateSync', () => {
+    it('validates a class instance correctly', () => {
+      const invitation = new ConnectionInvitationMessage({
+        did: 'did:sov:test1234',
+        id: 'afe2867e-58c3-4a8d-85b2-23370dd9c9f0',
+        label: 'test-label',
+      })
+
+      expect(MessageValidator.validateSync(invitation, ConnectionInvitationMessage)).toEqual([])
+    })
+    it('throws an error for invalid class instance', () => {
+      const invitation = new ConnectionInvitationMessage({
+        did: 'did:sov:test1234',
+        id: 'afe2867e-58c3-4a8d-85b2-23370dd9c9f0',
+        label: 'test-label',
+      })
+      invitation.did = 'jnsjcd'
+
+      expect(MessageValidator.validateSync(invitation, ConnectionInvitationMessage)).toThrow
```

Comment:
If the method is not sync it should always be wrapped, so I'm not sure if this is actually validating. If I change it to the following it is not actually throwing:

```suggestion
      expect(() => MessageValidator.validateSync(invitation, ConnectionInvitationMessage)).toThrow()
```

I don't think you can use `toThrow` without the `()`. Same with `.rejects` with nothing following it (should be `.rejects.toThrow()`

You can make the tests (all in this file are not correct) working by setting the `.did = undefined` this will trigger a validation error

Reply:
ah sorry - yes, i missunderstood that!😅